Yoast/yoast-components
# Sets static ids on elements close to the input fields

## Summary

Adds ID's to elements. For the purpose of testing via puppeteer.

* Adds the id "snippet-editor-field-title" to the inputcontainer, before draft.js is used.
* Adds the id "snippet-editor-field-slug" to the input element the slug is in.
* Adds the id "snippet-editor-field-description" to the inputcontainer, before draft.js is used.

**Branched from release/7.7, we should wait until the release of 7.7 before merging this one**

This PR can be summarized in the following changelog entry:

* N/A.

## Relevant technical choices:

* Sets the id for the slug at the input field of the slug.
* Sets the id "snippet-editor-field-title" and the id  on the inputContainer component, before draft.js is used.
* Edit Maarten: came across an oversight with the onBlur functionality which caused some weird behavior with clicking a focused field. Fixed by implementing the onBlur, and thereby accidentally fixing https://github.com/Yoast/yoast-components/issues/619 halfway.

## Test instructions

This PR can be tested by following these steps:
*NOTE* test together with https://github.com/Yoast/wordpress-seo/pull/10168
* Open a post page/ the standalone editor.
* Open the inspector.
* Check the input fields for the snippet editor. 
  * Check if the input fields for the title and description have their respective id's where the classname of the element contains: `Shared__InputContainer`
  * Check if the input field for the slug has his respective id.

Fixes https://github.com/Yoast/wordpress-seo/issues/10090
Fixes https://github.com/Yoast/yoast-components/issues/619

**Note that this fixes #619 only partially. Split out remaining issues** to https://github.com/Yoast/yoast-components/issues/624

Requires https://github.com/Yoast/wordpress-seo/issues/10091
